changelogs/fragments/1978-jira-transition-logic.yml

```
@@ -0,0 +1,4 @@
+bugfixes:
+  - jira - fixed fields' update in ticket transitions (https://github.com/ansible-collections/community.general/issues/818).
+minor_changes:
+  - jira - added parameter ``accountId`` for compatibility with recent versions of JIRA (https://github.com/ansible-collections/community.general/issues/818).
```

```suggestion
  - jira - added parameter ``accountId`` for compatibility with recent versions of JIRA (https://github.com/ansible-collections/community.general/pull/1978).
```
Or is the new option really related to #818?

---

I was in doubt here myself: it is related in an indirect way. 

I went that way because I was trying to change the `assignee` during the transition, per #818 but while investigating that I noticed that passing the username was not working in any other operation. That lead me into this issue with the accountId, so it does solve #818 as far as the assignee is concerned, but helps beyond that.

---

How about
```suggestion
  - jira - added parameter ``accountId`` for compatibility with recent versions of JIRA (https://github.com/ansible-collections/community.general/issues/818, https://github.com/ansible-collections/community.general/pull/1978).
```
then?

---

works for me :)